### PR TITLE
Create a heroku_space data resource.

### PR DESCRIPTION
**Pull request by joestump**
Inspired by the work in #41 here's a data resource for [Heroku Private Spaces](https://www.heroku.com/private-spaces).

Example HCL:

```hcl
provider "heroku" {}

data "heroku_space" "test" {
  name = "pcg-proof-of-concept"
}

output "space" {
  value = {
    state        = "${data.heroku_space.test.state}"
    shield       = "${data.heroku_space.test.shield}"
    name         = "${data.heroku_space.test.name}"
    organization = "${data.heroku_space.test.organization}"
  }
}
```

Example output: 

```
data.heroku_space.test: Refreshing state...

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

space = {
  name = my-secret-space
  organization = my-organization-name
  shield = false
  state = allocated
}
```

### CODE REVIEW

**Comment by joestump** [comment]
Updates:

* Updated the data source to align more closely with the work in #44. 
* Added a basic test.

**Comment by sheax0r** [comment]
@joestump hey joe! this branch has a conflict now. If you can resolve (it looks like a trivial conflict with the recent merge of the `heroku_app` datasource), i'll run the acceptance tests and merge as long as they pass.

File: website/docs/d/space.html.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+## Attributes Reference
+
+The following attributes are exported:
+
+* `name` - The name of the Heroku Private Space. In Heroku, this is also the unique .
+* `id` - The unique ID of the Heroku Private Space.
+* `region` - The region in which the Heroku Private Space is deployed.
+* `state` - The state of the Heroku Private Space. Either `allocating` or `allocated`.
+* `shielded` - Whether or not the space has [Shield](https://devcenter.heroku.com/articles/private-spaces#shield-private-spaces) turned on. One of `on` or `off`.
+* `organization` - The organization that owns this app, if the app is owned by an organization. The fields for this block are documented below.
```
Comment (sheax0r) [inline website/docs/d/space.html.markdown]: app --> space

**Comment by joestump** [comment]
@sheax0r I've appeased the linting gods; we're green again. :shipit:

**Comment by sheax0r** [comment]
@joestump I'm seeing this failure when I try to run the acceptance tests:

```
--- FAIL: TestAccDatasourceHerokuSpace_Basic (0.15s)
	testing.go:434: Step 0 error: Error applying: 1 error(s) occurred:

		* heroku_space.foobar: 1 error(s) occurred:

		* heroku_space.foobar: Post https://api.heroku.com/spaces: Private Space cannot be created in specified region.
	testing.go:494: Error destroying resource! WARNING: Dangling resources
		may exist. The full state and error is shown below.

		Error: Configuration is invalid.

		Warnings: []string(nil)

		Errors: []string{"1 error(s) occurred:\n\n* module root: 1 error(s) occurred:\n\n* resource 'data.heroku_app.foobar' config: unknown resource 'heroku_app.foobar' referenced in variable heroku_app.foobar.name"}

		State: <no state>
```

I believe that the "us" region is not valid for creating spaces - "virginia" would probably work, I believe that's what I used in space-related tests for the heroku_app datasource.

Have you successfully run this test yourself?

Btw there are [two other failing tests related to spaces](https://github.com/terraform-providers/terraform-provider-heroku/issues/49), but this one isn't one of those (it's part of this PR).

**Comment by joestump** [comment]
@sheax0r @bernerdschaefer I think this is good to go now that #51 is merged. The last failing test appears to be exercising renaming a space and was failing on `master` when I ran them.

**Pull request merged**